leakcanary-analyzer/src/main/java/com/squareup/leakcanary/HeapAnalyzer.java

```diff
@@ -90,6 +95,28 @@ public AnalysisResult checkForLeak(File heapDumpFile, String referenceKey) {
     }
   }
 
+  void dedupGcRoots(Snapshot snapshot) {
+    // THashMap has smaller footprint than HashMap
+    final THashMap<String, RootObj> dedupMap = new THashMap<>();
```

Comment:
THashMap doesn't expose that capability.  Its superclass THash sets a default capacity internally.